app/src/main/java/org/schabi/newpipe/util/potoken/PoTokenWebView.kt

```diff
@@ -0,0 +1,356 @@
+package org.schabi.newpipe.util.potoken
+
+import android.content.Context
+import android.os.Build
+import android.os.Handler
+import android.os.Looper
+import android.util.Log
+import android.webkit.JavascriptInterface
+import android.webkit.WebView
+import androidx.annotation.MainThread
+import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
+import io.reactivex.rxjava3.core.Single
+import io.reactivex.rxjava3.core.SingleEmitter
+import io.reactivex.rxjava3.disposables.CompositeDisposable
+import io.reactivex.rxjava3.schedulers.Schedulers
+import org.schabi.newpipe.BuildConfig
+import org.schabi.newpipe.DownloaderImpl
+import java.time.Instant
+
+class PoTokenWebView private constructor(
+    context: Context,
+    // to be used exactly once only during initialization!
+    private val generatorEmitter: SingleEmitter<PoTokenGenerator>,
+) : PoTokenGenerator {
+    private val webView = WebView(context)
+    private val disposables = CompositeDisposable() // used only during initialization
+    private val poTokenEmitters = mutableListOf<Pair<String, SingleEmitter<String>>>()
```

Comment:
```suggestion
    private val poTokenEmitters = ConcurrentHashMap<String, SingleEmitter<String>>()
```

Reply:
No, a list here is totally fine. Small lists are much faster that hash maps and I don't expect this list to grow bigger than a couple of items.